tokio/src/runtime/context.rs

```diff
@@ -292,11 +268,15 @@ cfg_rt! {
         DisallowBlockInPlaceGuard(reset)
     }
 
-    pub(crate) fn with_defer<R>(f: impl FnOnce(&mut Defer) -> R) -> Option<R> {
-        CONTEXT.with(|c| {
-            let mut defer = c.defer.borrow_mut();
-            defer.as_mut().map(f)
-        })
+    #[track_caller]
+    pub(crate) fn defer(waker: &Waker) {
+        with_scheduler(|maybe_scheduler| {
+            if let Some(scheduler) = maybe_scheduler {
+                scheduler.defer(waker);
+            } else {
+                waker.wake_by_ref();
```

Comment:
We should also `wake_by_ref` if the thread local is destroyed due to thread shutdown.

Reply:
This is the case when it is called from outside the runtime.

Reply:
I added a comment and a test checking `yield_now()` from outside the runtime.